hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -560,7 +570,14 @@ protected String removeListenerLocally(CacheEntryListenerConfiguration<K, V> cac
         } else {
             regs = asyncListenerRegistrations;
         }
-        return regs.remove(cacheEntryListenerConfiguration);
+        String registrationId = regs.remove(cacheEntryListenerConfiguration);
+        if (registrationId != null) {
+            Closeable closeable = closeableListeners.remove(registrationId);
+            if (closeable != null) {
```

Comment:
That `null` should not be necessary:
```
    public static void closeResource(Closeable closeable) {
        if (closeable == null) {
            return;
        }
```
So you can remove it to avoid code coverage issues for the `is null` case :)